### PR TITLE
C++: Promote `cpp/incorrect-allocation-error-handling`

### PR DESCRIPTION
**Pull request by MathiasVP**
This PR moves `cpp/incorrect-allocation-error-handling` out of experimental with `@precision medium`. This means it won't go into the Code Scanning suite yet, nor will it be displayed by default on LGTM. It will, however, be run on LGTM. This will give us some helpful data on whether we can safely move this to a higher precision.

<b>Note:</b> The documentation for this query was improved in https://github.com/github/codeql/pull/5804, but it has not received a docs review yet. I made some conscious whitespace changes in the first commit (95045929096f4b96eaedefb816ed42515b079583) to help @github/docs-content-codeql review it (I hope that works 🤞 ).

The results on LGTM look good: https://lgtm.com/query/600361054747246847/

### CODE REVIEW

**Comment by MathiasVP** [comment]
@geoffw0 I've looked at the result on `vxWidgets/wxWidgets` that we believed was a false positive [here](https://github.com/github/codeql/pull/5847#issuecomment-834492309), but it turns out to be a true positive:

It turns out the null check is down here:
```cpp
    wxLogTrace( FILECONF_TRACE_MASK,
                wxT("        tail: %s"),
                ((m_linesTail) ? (const wxChar*)m_linesTail->Text().c_str()
                               : wxEmptyString) );
```
where there's an alert on the `(m_linesTail) ?` part. This alert is a TP because there's an assignment further up like so:
```cpp
wxFileConfigLineList *pLine = new wxFileConfigLineList(str);
/*...*/
m_linesTail = pLine;
```

**Review by geoffw0 — DISMISSED** [review]
> The results on LGTM look good: https://lgtm.com/query/600361054747246847/

I agree, results look great (and the bar for `@precision medium` is not so high anyway).

**Comment by geoffw0** [comment]
Do we want to wait for feedback from the documentation team, or make any change requests as follow-up?

**Comment by MathiasVP** [comment]
> Do we want to wait for feedback from the documentation team, or make any change requests as follow-up?

I think we should wait for the documentation team. The next dist upgrade is still 11 days away, so we can afford to wait some time before merging this.

**Comment by felicitymay** [comment]
@MathiasVP, this week's docs content team's first responder here👋🏻

Thanks for the ping and for being prepared to wait for the docs team to review this pull request. I'll add this to our review board and we'll aim to review within 2 business days. It sounds as if that should be okay for your deadline.

**Comment by MathiasVP** [comment]
> Thanks for the ping and for being prepared to wait for the docs team to review this pull request. I'll add this to our review board and we'll aim to review within 2 business days. It sounds as if that should be okay for your deadline.

Thanks for the heads up. That sounds absolutely fine!

**Comment by mchammer01** [comment]
Commenting here as I can't comment on the diffs:

`ql` file - if at all possible, the description should follow the pattern described in the [Query metadata style guide](https://github.com/github/codeql/blob/main/docs/query-metadata-style-guide.md#query-descriptions-description).  Perhaps update to something into these lines: `Mixing up the failure conditions `operator new` exception on allocation with the `operator new(std::nothrow)` null pointer can result in unexpected behavior.'? 
I am not an engineer/developer so feel free to rewrite if this doesn't make any sense 😉 
Reviewing the qhelp file now.

**Comment by MathiasVP** [comment]
@mchammer01 you should be able to do inline review comments if you view the changes in 95045929096f4b96eaedefb816ed42515b079583.

**Comment by mchammer01** [comment]
Oh thank you 🤦🏻

**Review by mchammer01 — APPROVED** [review]
@MathiasVP - this LGTM ✨ 
I'm approving this as I don't want to block you, and there are only a couple of minor issues.
Would be great if you could address my review comments (query description and typo fix).
Hope this helps!

**Review by geoffw0 — APPROVED** [review]
I'm happy, I believe the docs team are happy, I'll merge this now.

File: cpp/ql/src/Security/CWE/CWE-570/IncorrectAllocationErrorHandling.ql
Comment (geoffw0) [inline cpp/ql/src/Security/CWE/CWE-570/IncorrectAllocationErrorHandling.ql:3]: :+1:

**Pull request merged**
